tests/checkpointing/test_model_checkpoint.py

```diff
@@ -986,3 +986,43 @@ def __init__(self, hparams):
     else:
         # make sure it's not AttributeDict
         assert type(ckpt[model.CHECKPOINT_HYPER_PARAMS_KEY]) == hparams_type
+
+
+@pytest.mark.parametrize(
+    'save_top_k, expected',
+    [
+        (1, ['curr_epoch.ckpt']),
+        (2, ['curr_epoch.ckpt', 'curr_epoch-v0.ckpt']),
+    ]
+)
+def test_model_checkpoint_file_already_exists(tmpdir, save_top_k, expected):
+    """
+    Test that version is added to filename if required and it already exists in dirpath.
+    """
+    class CustomModel(LogInTwoMethods):
+        def validation_epoch_end(self, outputs):
+            return {'epoch': self.current_epoch}
+
+    model_checkpoint = ModelCheckpoint(
+        dirpath=tmpdir,
+        filename='curr_epoch',
+        save_top_k=save_top_k,
+        monitor='epoch',
+        mode='max',
+    )
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        callbacks=[model_checkpoint],
+        max_epochs=4,
```

Comment:
can you parametrize this test for epochs (2, 3)? The person who reported the issue claimed:

> from one run to another i might get as a final model either the model with v0  or the one without?

Reply:
done.